### PR TITLE
fix(datagrid): expanded row keyboard navigation

### PR DESCRIPTION
**Pull request by valentin-mladenov**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?
Expanded row is inaccessible by keyboard navigation.

Issue Number: CDE-2200

## What is the new behavior?
Expanded row is accessible by keyboard navigation.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: projects/angular/src/data/datagrid/datagrid-row-detail.ts [hunk 1]
```diff
@@ -67,16 +76,14 @@ export class ClrDatagridRowDetail implements AfterContentInit, OnDestroy {
   get beginningOfExpandableContentAriaText() {
     return (
       this._beginningOfExpandableContentAriaText ||
-      `${this.commonStrings.keys.datagridExpandableBeginningOf} 
-      ${this.commonStrings.keys.datagridExpandableRowContent}`
+      `${this.commonStrings.keys.datagridExpandableBeginningOf} ${this.commonStrings.keys.datagridExpandableRowContent}`
```
[the remark this answers is not in the document]
Comment (kevinbuhmann) [reply projects/angular/src/data/datagrid/datagrid-row-detail.ts:79]: Same with the instance below in `endOfExpandableContentAriaText`. It is not obvious when reading the changes that both actually changed here.

[the remark this answers is not in the document]
Comment (valentin-mladenov) [reply projects/angular/src/data/datagrid/datagrid-row-detail.ts]: @kevinbuhmann moved it to separate PR #1708, because IDE overrrided it again.

File: projects/demo/src/app/datagrid/density/density.html [hunk 2]
```diff
@@ -52,6 +52,10 @@ <h2>Dense Datagrid</h2>
       <input type="checkbox" clrCheckbox name="remove-row-detail" [(ngModel)]="removeRowDetail" />
       <label>Remove Row Detail</label>
     </clr-checkbox-wrapper>
+    <clr-checkbox-wrapper>
+      <input type="checkbox" clrCheckbox name="remove-row-detail" [(ngModel)]="showAnchor" />
```
Comment (Jinnie) [inline projects/demo/src/app/datagrid/density/density.html:56]: the name should be different

Comment (valentin-mladenov) [reply projects/demo/src/app/datagrid/density/density.html]: corrected in ad0607f

**Review by Jinnie — APPROVED** [review]
Thanks for handling all small corner cases we discussed.
Looks good to me. Tests are adequate and easy to add to.
